ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -72,33 +72,40 @@ public CopyUtils(String distCpDoAsUser, HiveConf hiveConf) {
   public void copyAndVerify(FileSystem destinationFs, Path destRoot,
                     List<ReplChangeManager.FileInfo> srcFiles) throws IOException, LoginException, HiveFatalException {
     Map<FileSystem, Map< Path, List<ReplChangeManager.FileInfo>>> map = fsToFileMap(srcFiles, destRoot);
-    for (Map.Entry<FileSystem, Map<Path, List<ReplChangeManager.FileInfo>>> entry : map.entrySet()) {
-      FileSystem sourceFs = entry.getKey();
-      Map<Path, List<ReplChangeManager.FileInfo>> destMap = entry.getValue();
-      for (Map.Entry<Path, List<ReplChangeManager.FileInfo>> destMapEntry : destMap.entrySet()) {
-        Path destination = destMapEntry.getKey();
-        List<ReplChangeManager.FileInfo> fileInfoList = destMapEntry.getValue();
-        boolean useRegularCopy = regularCopy(destinationFs, sourceFs, fileInfoList);
-
-        if (!destinationFs.exists(destination)
-                && !FileUtils.mkdir(destinationFs, destination, hiveConf)) {
-          LOG.error("Failed to create destination directory: " + destination);
-          throw new IOException("Destination directory creation failed");
-        }
+    UserGroupInformation proxyUser = getProxyUser();
+    try {
+      for (Map.Entry<FileSystem, Map<Path, List<ReplChangeManager.FileInfo>>> entry : map.entrySet()) {
+        FileSystem sourceFs = entry.getKey();
```

Comment:
The sourceFs should be re-get from HDFS if previous copy closed all file systems object for proxy user.

Reply:
The hive retry logic is for very limited local functionality ..for all other places ..in case of file system error ..dlm will retry